runtime error: index out of range

panic: runtime error: index out of range

goroutine 1 [running]:
go_sadb/src/github.com/lsl/sadb.select_devices(0xc0000c00a0, 0x2, 0x2, 0x11c62b8, 0x11c62b8, 0x11c66d0)
	/Users/lishoulin/Desktop/go/go_sadb/src/github.com/lsl/sadb/sadb.go:46 +0x6db
go_sadb/src/github.com/lsl/sadb.Sadb()
	/Users/lishoulin/Desktop/go/go_sadb/src/github.com/lsl/sadb/sadb.go:119 +0xcf
main.main()
	/Users/lishoulin/Desktop/go/go_sadb/main.go:6 +0x20

--- a/src/github.com/lsl/gadb/gadb.go
+++ b/src/github.com/lsl/gadb/gadb.go
@@ -39,11 +39,11 @@ func selectDevices(devs []devices) []devices {
 	default:
 		var arrays []devices
 		var c, err = strconv.Atoi(line)
-		if err != nil || c < 0 || c > count {
+		if err != nil || c < 0 || c-1 > count {
 			fmt.Printf("error input: %s, retry again\n", line)
 			return selectDevices(devs)
 		}
-		arrays = append(arrays, devs[c])
+		arrays = append(arrays, devs[c-1])
 		return arrays
 	}
 	return nil

